packages/core/src/local-admin-api.ts

```diff
@@ -1,26 +1,70 @@
 import { AdminApi, DiagnosticsLogger, LogStyle } from '@ceramicnetwork/common'
 import { StreamID } from '@ceramicnetwork/streamid'
 import { LocalIndexApi } from './indexing/local-index-api.js'
+import crypto from 'crypto'
+
+// TODO: Expose this as a config option?
+const ADMIN_API_AUTHORIZATION_TIMEOUT = 1000 * 60 * 1 // 1 min
+
+type AdminCode = string
+type Timestamp = number
+
+type AdminCodeCache = {
+  [key: AdminCode]: Timestamp
+}
 
 /**
  * AdminApi for Ceramic core.
  */
 export class LocalAdminApi implements AdminApi {
+  private readonly codeCache: AdminCodeCache = {} as AdminCodeCache
+
   constructor(
+    private readonly adminDids: Array<string> | undefined,
     private readonly indexApi: LocalIndexApi,
     private readonly logger: DiagnosticsLogger
   ) {}
 
-  async startIndexingModels(modelsIDs: Array<StreamID>): Promise<void> {
+  private verifyActingDid(actingDid: string) {
+    if (!this.adminDids || !this.adminDids.some( adminDid => actingDid.startsWith(adminDid) )) {
+      this.logger.log(LogStyle.warn, `Unauthorized access attempt to Admin Api from did: ${actingDid}`)
+      throw Error(`Unauthorized access: DID '${actingDid}' does not have admin access permission to this node`)
+    }
+  }
+
+  async generateCode(): Promise<string> {
+    const newCode = crypto.randomUUID()
+    const now = (new Date).getTime()
+    this.codeCache[newCode] = now
```

Comment:
I would probably just move the `codeCacahe` into the `ceramic-daemon.ts` file, but it doesn't matter that much.